### PR TITLE
BUG: tz aware Timestamp field accessors returns local values (#13303)

### LINKED ISSUE #13303: Timezone aware Timestamp.dayofyear does not report correct day
#### Code Sample, a copy-pastable example if possible

```
import pandas as PD

time_str = '2015-10-01 23:59:00-04:00'
pd_dto = PD.to_datetime(time_str, utc=True).tz_convert('US/Eastern')

actual_yday = pd_dto.date().timetuple().tm_yday
print 'pandas thinks', pd_dto, 'is the', pd_dto.dayofyear, 'yday'
print 'when really its the', actual_yday
```

#### Expected Output

I expect the dayofyear to agree with tm_yday, but they do not.

#### output of `pd.show_versions()`

<details>
## INSTALLED VERSIONS

commit: None
python: 2.7.10.final.0
python-bits: 64
OS: Windows
OS-release: 8
machine: AMD64
processor: Intel64 Family 6 Model 61 Stepping 4, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None

pandas: 0.16.2
nose: 1.3.7
Cython: 0.22.1
numpy: 1.9.2
scipy: 0.15.1
statsmodels: 0.6.1
IPython: 3.2.0
sphinx: 1.3.1
patsy: 0.3.0
dateutil: 2.4.2
pytz: 2015.4
bottleneck: 1.0.0
tables: 3.2.0
numexpr: 2.4.3
matplotlib: 1.4.3
openpyxl: 1.8.5
xlrd: 0.9.3
xlwt: 1.0.0
xlsxwriter: 0.7.3
lxml: 3.4.4
bs4: 4.3.2
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.0.5
pymysql: None
psycopg2: None
</details>

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #13303
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master | flake8 --diff``
 - [x] whatsnew entry

Previously, calling a date/time attribute with Timestamp that's tz aware (e.g. `Timestamp('...', tz='...').dayofyear`) would return the attribute in UTC instead of the local tz.

### CODE REVIEW

File: pandas/_libs/tslib.pyx [hunk 1]
```diff
@@ -1233,7 +1233,10 @@ cdef class _Timestamp(datetime):
...
 
     cpdef _get_field(self, field):
-        out = get_date_field(np.array([self.value], dtype=np.int64), field)
+        val = self.value
+        if self.tz is not None and not _is_utc(self.tz):
+            val = tz_convert_single(self.value, 'UTC', self.tz)
+        out = get_date_field(np.array([val], dtype=np.int64), field)
         return int(out[0])
 
     cpdef _get_start_end_field(self, field):
```
Comment (jorisvandenbossche) [inline pandas/_libs/tslib.pyx]: Didn't check, but does this `_get_start_end_field` needs the same ?

**Review by jorisvandenbossche — COMMENTED** [review]
Looks good! 
Small comment

File: pandas/_libs/tslib.pyx (hunk 1 above)
Comment (jreback) [inline pandas/_libs/tslib.pyx]: this is ok, get_date_field handles an i8 only.

**Comment by jreback** [comment]
can you add similar type tests for ``DatetimeIndex`` (IOW with a tz) in ``pandas/tests/indexes/datetimes/test_misc.py``

File: pandas/_libs/tslib.pyx (hunk 1 above)
Comment (jorisvandenbossche) [inline pandas/_libs/tslib.pyx]: Not sure what you mean with `get_date_field`, but this example seems to be wrong:

```
In [3]: pd.Timestamp("2014-01-01 00:00:00+01:00").is_month_start
Out[3]: False

In [4]: pd.Timestamp("2014-01-01 00:00:00").is_month_start
Out[4]: True
```

Comment (jreback) [inline pandas/_libs/tslib.pyx]: ahh sorry, was looking at something else. ``get_start_end_field`` looks to need fixing as well. thanks @jorisvandenbossche

**Comment by jreback** [comment]
lgtm.

**Comment by mroeschke** [comment]
Thanks @jreback!

**Pull request closed without merging**